migrations/800_groups__add.sql

```diff
@@ -0,0 +1,30 @@
+-- A Work in Progress to discuss shared schema for group storage in PL
+
+CREATE TABLE IF NOT EXISTS group_configs (
+    id BIGSERIAL PRIMARY KEY,
+    course_instance_id BIGINT NOT NULL REFERENCES course_instances(id) ON DELETE CASCADE ON UPDATE CASCADE,
+    assessment_id BIGINT REFERENCES assessments(id) ON DELETE CASCADE ON UPDATE CASCADE,
+    type TEXT, -- what will these be? This might belong in groups rather than groups_config
+    name TEXT, -- or maybe description is accurate here? Maybe as a base for groups created from it?
+    minimum INT,
+    maximum INT,
+    date timestamp with time zone DEFAULT CURRENT_TIMESTAMP,
+    deleted_at timestamp with time zone
+);
+
+CREATE TABLE IF NOT EXISTS groups (
+    id BIGSERIAL PRIMARY KEY,
+    course_instance_id BIGINT NOT NULL REFERENCES course_instances(id) ON DELETE CASCADE ON UPDATE CASCADE,
+    name TEXT, -- visible name of the group
+    group_config_id BIGINT REFERENCES group_configs(id) ON DELETE CASCADE ON UPDATE CASCADE,
+    date timestamp with time zone DEFAULT CURRENT_TIMESTAMP,
+    deleted_at timestamp with time zone,
+    UNIQUE (course_instance_id, name)
```

Comment:
I used a different approach in the other PR for this. Might want to keep that in sync (and since your code is the one actively using it, use your stuff to try out that PR's constraints?)